source/extensions/filters/http/grpc_stats/grpc_stats_filter.cc

```diff
@@ -154,6 +154,16 @@ class GrpcStatsFilter : public Http::PassThroughFilter {
           // Get dynamically-allocated Context::RequestStatNames from the context.
           request_names_ = config_->context_.resolveDynamicServiceAndMethod(headers.Path());
           do_stat_tracking_ = request_names_.has_value();
+          if (do_stat_tracking_) {
```

Comment:
Have you thought of changing `DynamicName` to store an actual copy of a string instead of a string_view? This would prevent this issue from occurring in the future...

Reply:
Do you mean change
```
explicit DynamicName(absl::string_view str) : absl::string_view(str) {}
```
to
```
explicit DynamicName(std::string str) : std::string(str) {}
```

or make it based on `std::string`?

I am afraid that we need `string_view` in some cases like:
```
Utility::counterFromElements(*scope, {DynamicName("a"), DynamicName("b")});
```

Reply:
@dmitri-d could you share more details?

Reply:
Maybe we create another class, based on `std::string`? so that now we have 2 classes **StatName** which internalizes values, **DynamicName** holding the memory reference and one more which stores the actual data.

BTW, can someone explain why we cannot use **StatName** here? I tried but got a crash on parallel execution - probably I did something wrong